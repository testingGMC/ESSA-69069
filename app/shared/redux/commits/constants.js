/*
 *
 * Commits constants
 *
 */

export const FETCH_COMMITS_REQUEST = 'app/Commits/FETCH_COMMITS_REQUEST'
export const FETCH_COMMITS_SUCCESS = 'app/Commits/FETCH_COMMITS_SUCCESS'
export const FETCH_COMMITS_FAILURE = 'app/Commits/FETCH_COMMITS_FAILURE'

export const FETCH_LATEST_COMMITS_REQUEST = 'app/Commits/FETCH_LATEST_COMMITS_REQUEST'
export const FETCH_LATEST_COMMITS_SUCCESS = 'app/Commits/FETCH_LATEST_COMMITS_SUCCESS'
export const FETCH_LATEST_COMMITS_FAILURE = 'app/Commits/FETCH_LATEST_COMMITS_FAILURE'

export const REVERT_COMMIT_REQUEST = 'app/Commits/REVERT_COMMIT_REQUEST'
export const REVERT_COMMIT_SUCCESS = 'app/Commits/REVERT_COMMIT_SUCCESS'
export const REVERT_COMMIT_FAILURE = 'app/Commits/REVERT_COMMIT_FAILURE'