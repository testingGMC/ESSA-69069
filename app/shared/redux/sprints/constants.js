/*
 *
 * Sprints constants
 *
 */

export const FETCH_SPRINTS_FIRST_TIME_REQUEST =
  'app/Sprints/FETCH_SPRINTS_FIRST_TIME_REQUEST'
export const FETCH_SPRINTS_FIRST_TIME_SUCCESS =
  'app/Sprints/FETCH_SPRINTS_FIRST_TIME_SUCCESS'
export const FETCH_SPRINTS_FIRST_TIME_FAILURE =
  'app/Sprints/FETCH_SPRINTS_FAILURE'

export const ADD_SPRINT_FIRST_TIME_REQUEST =
  'app/Sprints/ADD_SPRINT_FIRST_TIME_REQUEST'
export const ADD_SPRINT_FIRST_TIME_SUCCESS =
  'app/Sprints/ADD_SPRINT_FIRST_TIME_SUCCESS'
export const ADD_SPRINT_FIRST_TIME_FAILURE = 'app/Sprints/ADD_SPRINT_FAILURE'

export const FETCH_SPRINTS_REQUEST = 'app/Sprints/FETCH_SPRINTS_REQUES3T'
export const FETCH_SPRINTS_SUCCESS = 'app/Sprints/FETCH_SPRINTS_SUCCESS'
export const FETCH_SPRINTS_FAILURE = 'app/Sprints/FETCH_SPRINTS_FAILURE'

export const FETCH_WORK_ITEMS_REQUEST = 'app/Sprints/FETCH_WORK_ITEMS_REQUEST'
export const FETCH_WORK_ITEMS_SUCCESS = 'app/Sprints/FETCH_WORK_ITEMS_SUCCESS'
export const FETCH_WORK_ITEMS_FAILURE = 'app/Sprints/FETCH_WORK_ITEMS_FAILURE'

export const ADD_SPRINT_REQUEST = 'app/Sprints/ADD_SPRINT_REQUEST'
export const ADD_SPRINT_SUCCESS = 'app/Sprints/ADD_SPRINT_SUCCESS'
export const ADD_SPRINT_FAILURE = 'app/Sprints/ADD_SPRINT_FAILURE'

export const ADD_WORK_ITEM_REQUEST = 'app/Sprints/ADD_WORK_ITEM_REQUEST'
export const ADD_WORK_ITEM_SUCCESS = 'app/Sprints/ADD_WORK_ITEM_SUCCESS'
export const ADD_WORK_ITEM_FAILURE = 'app/Sprints/ADD_WORK_ITEM_FAILURE'

export const UPDATE_WORK_ITEM_REQUEST = 'app/Sprints/UPDATE_WORK_ITEM_REQUEST'
export const UPDATE_WORK_ITEM_SUCCESS = 'app/Sprints/UPDATE_WORK_ITEM_SUCCESS'
export const UPDATE_WORK_ITEM_FAILURE = 'app/Sprints/UPDATE_WORK_ITEM_FAILURE'

export const UPDATE_WORK_ITEM_STATE_REQUEST =
  'app/Sprints/UPDATE_WORK_ITEM_STATE_REQUEST'
export const UPDATE_WORK_ITEM_STATE_SUCCESS =
  'app/Sprints/UPDATE_WORK_ITEM_STATE_SUCCESS'
export const UPDATE_WORK_ITEM_STATE_FAILURE =
  'app/Sprints/UPDATE_WORK_ITEM_FAILURE'

export const DELETE_WORK_ITEM_REQUEST = 'app/Sprints/DELETE_WORK_ITEM_REQUEST'
export const DELETE_WORK_ITEM_SUCCESS = 'app/Sprints/DELETE_WORK_ITEM_SUCCESS'
export const DELETE_WORK_ITEM_FAILURE = 'app/Sprints/DELETE_WORK_ITEM_FAILURE'

export const SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_REQUEST =
  'app/Sprints/SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_REQUEST'
export const SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_SUCCESS =
  'app/Sprints/SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_SUCCESS'
export const SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_FAILURE =
  'app/Sprints/SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_FAILURE'

export const SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_REQUEST =
  'app/Sprints/SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_REQUEST'
export const SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_SUCCESS =
  'app/Sprints/SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_SUCCESS'
export const SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_FAILURE =
  'app/Sprints/SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_FAILURE'

export const SET_IS_ADD_SPRINT_MODAL_VISIBLE_REQUEST =
  'app/Sprints/SET_IS_ADD_SPRINT_MODAL_VISIBLE_REQUEST'
export const SET_IS_ADD_SPRINT_MODAL_VISIBLE_SUCCESS =
  'app/Sprints/SET_IS_ADD_SPRINT_MODAL_VISIBLE_SUCCESS'
export const SET_IS_ADD_SPRINT_MODAL_VISIBLE_FAILURE =
  'app/Sprints/SET_IS_ADD_SPRINT_MODAL_VISIBLE_FAILURE'

export const CLEAR_ERRORS = 'app/Sprints/CLEAR_ERRORS'
