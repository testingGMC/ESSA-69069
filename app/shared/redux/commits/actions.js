/*
 *
 * Commits actions
 *
 */
import * as CONSTANTS from './constants'

export function fetchCommits() {
  return {
    type: CONSTANTS.FETCH_COMMITS_REQUEST,
  }
}

export function fetchLatestCommits() {
  return {
    type: CONSTANTS.FETCH_LATEST_COMMITS_REQUEST,
  }
}

export function revertToACommit(id, message, description) {
  return {
    type: CONSTANTS.REVERT_COMMIT_REQUEST,
    id,
    message,
    description,
  }
}
