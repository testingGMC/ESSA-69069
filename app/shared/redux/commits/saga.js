import { put, takeEvery, all, call } from 'redux-saga/effects'
import * as api from 'shared/services/version-control.service'
import { push } from 'connected-react-router'
import * as CONSTANTS from './constants'

export function* fetchCommits() {
  try {
    const data = yield call(api.getCommits)

    yield put({
      type: CONSTANTS.FETCH_COMMITS_SUCCESS,
      commits: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_COMMITS_FAILURE, e })
  }
}
export function* fetchLatestCommits() {
  try {
    const data = yield call(api.getLatestCommits)
    yield put({
      type: CONSTANTS.FETCH_LATEST_COMMITS_SUCCESS,
      commits: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_LATEST_COMMITS_FAILURE, e })
  }
}
export function* revertToACommit(action) {
  try {
    const data = yield call(
      api.revertToACommit,
      action.id,
      action.message,
      action.description,
    )
    yield put({
      type: CONSTANTS.REVERT_COMMIT_SUCCESS,
      commit: data.model,
    })
    yield put(push('/'))
  } catch (e) {
    yield put({ type: CONSTANTS.REVERT_COMMIT_FAILURE, e })
  }
}

export function* fetchCommitsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_COMMITS_REQUEST, fetchCommits)
}

export function* fetchLatestCommitsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_LATEST_COMMITS_REQUEST, fetchLatestCommits)
}

export function* revertCommitsWatcher() {
  yield takeEvery(CONSTANTS.REVERT_COMMIT_REQUEST, revertToACommit)
}

export default function* commitsSaga() {
  yield all([
    fetchCommitsWatcher(),
    fetchLatestCommitsWatcher(),
    revertCommitsWatcher(),
  ])
}
