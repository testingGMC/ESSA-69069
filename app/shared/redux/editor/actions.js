/*
 *
 * Editor actions
 *
 */
import * as CONSTANTS from './constants'

export function clearState() {
  return {
    type: CONSTANTS.CLEAR_STATE_SUCCESS,
  }
}

export function fetchSkill(skillId) {
  return {
    type: CONSTANTS.FETCH_SKILL_REQUEST,
    skillId,
  }
}
export function updateSkill(skillId, skill) {
  return {
    type: CONSTANTS.UPDATE_SKILL_REQUEST,
    skillId,
    skill,
  }
}

export function addLearningStyle(learningStyle) {
  return {
    type: CONSTANTS.ADD_LEARNING_STYLE_REQUEST,
    learningStyle,
  }
}

export function deleteLearningStyle(learningStyleId) {
  return {
    type: CONSTANTS.DELETE_LEARNING_STYLE_REQUEST,
    learningStyleId,
  }
}

export function addPage(page, learningStyleId) {
  return {
    type: CONSTANTS.ADD_PAGE_REQUEST,
    page,
    learningStyleId,
  }
}

export function updatePage(pageId, page, learningStyleId) {
  return {
    type: CONSTANTS.UPDATE_PAGE_REQUEST,
    pageId,
    page,
    learningStyleId,
  }
}

export function deletePage(pageId, learningStyleId) {
  return {
    type: CONSTANTS.DELETE_PAGE_REQUEST,
    pageId,
    learningStyleId,
  }
}

export function addAssessment(assessment) {
  return {
    type: CONSTANTS.ADD_ASSESSMENT_REQUEST,
    assessment,
  }
}

export function deleteAssessment(assessmentId) {
  return {
    type: CONSTANTS.DELETE_ASSESSMENT_REQUEST,
    assessmentId,
  }
}

export function updateSelectedAssessmentId(assessmentId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_ASSESSMENT_ID_REQUEST,
    assessmentId,
  }
}

export function fetchTemplates() {
  return {
    type: CONSTANTS.FETCH_TEMPLATES_REQUEST,
  }
}

export function addTemplate(template) {
  return {
    type: CONSTANTS.ADD_TEMPLATE_REQUEST,
    template,
  }
}

export function deleteTemplate(templateId) {
  return {
    type: CONSTANTS.DELETE_TEMPLATE_REQUEST,
    templateId,
  }
}

export function updateTemplate(templateId, template) {
  return {
    type: CONSTANTS.UPDATE_TEMPLATE_REQUEST,
    templateId,
    template,
  }
}

export function updateSelectedLearningStyleId(learningStyleId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_LEARNING_STYLE_ID_REQUEST,
    learningStyleId,
  }
}
export function updateSelectedPageId(pageId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_PAGE_ID_REQUEST,
    pageId,
  }
}

export function addKeyword(keyword) {
  return {
    type: CONSTANTS.ADD_KEYWORD_REQUEST,
    keyword,
  }
}

export function updateKeyword(keywordId, keyword) {
  return {
    type: CONSTANTS.UPDATE_KEYWORD_REQUEST,
    keywordId,
    keyword,
  }
}

export function deleteKeyword(keywordId) {
  return {
    type: CONSTANTS.DELETE_KEYWORD_REQUEST,
    keywordId,
  }
}

export function updateSelectedKeywordId(keywordId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_KEYWORD_ID_REQUEST,
    keywordId,
  }
}

export function updateReorderChallengeQuestion(questionId, text) {
  return {
    type: CONSTANTS.UPDATE_REORDER_CHALLENGE_QUESTION_REQUEST,
    questionId,
    text,
  }
}

export function addReorderChallengeStatement(statement, assessmentId) {
  return {
    type: CONSTANTS.ADD_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statement,
    assessmentId,
  }
}

export function updateReorderChallengeStatement(
  statementId,
  statement,
  assessmentId,
) {
  return {
    type: CONSTANTS.UPDATE_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statementId,
    statement,
    assessmentId,
  }
}

export function deleteReorderChallengeStatement(statementId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statementId,
    assessmentId,
  }
}

export function updateCodeCompilingQuestion(questionId, question) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_QUESTION_REQUEST,
    questionId,
    text: question,
  }
}

export function updateCodeCompilingLanguage(languageId, language) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_LANGUAGE_REQUEST,
    languageId,
    text: language,
  }
}

export function updateCodeCompilingSolution(solutionId, solution) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_SOLUTION_REQUEST,
    solutionId,
    text: solution,
  }
}

export function updateCodeCompilingPlaceholder(placeholderId, placeholder) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_PLACEHOLDER_REQUEST,
    placeholderId,
    text: placeholder,
  }
}

export function updateCodeCompilingAddUseCase(assessmentId) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_ADD_TEST_CASE_REQUEST,
    assessmentId,
  }
}

export function updateCodeCompilingDeleteUseCase(testCaseId, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_DELETE_TEST_CASE_REQUEST,
    testCaseId,
    assessmentId,
  }
}

export function updateCodeCompilingUpdateUseCase(
  testCaseId,
  assessmentId,
  testCase,
) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_UPDATE_TEST_CASE_REQUEST,
    testCaseId,
    assessmentId,
    testCase,
  }
}

export function addQuizQuestion(question, assessmentId) {
  return {
    type: CONSTANTS.ADD_QUIZ_QUESTION_REQUEST,
    question,
    assessmentId,
  }
}

export function updateQuizQuestion(questionId, question, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_QUESTION_REQUEST,
    questionId,
    question,
    assessmentId,
  }
}

export function deleteQuizQuestion(questionId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_QUIZ_QUESTION_REQUEST,
    questionId,
    assessmentId,
  }
}

export function updateQuizLanguage(languageId, language) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_LANGUAGE_REQUEST,
    languageId,
    text: language,
  }
}

export function updateQuizCode(codeId, code) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_CODE_REQUEST,
    codeId,
    text: code,
  }
}

export function addQuizChoice(choice, questionId, assessmentId) {
  return {
    type: CONSTANTS.ADD_QUIZ_CHOICE_REQUEST,
    choice,
    questionId,
    assessmentId,
  }
}

export function updateQuizChoice(choiceId, choice, questionId) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_CHOICE_REQUEST,
    choiceId,
    choice,
    questionId,
  }
}

export function deleteQuizChoice(choiceId, questionId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_QUIZ_CHOICE_REQUEST,
    choiceId,
    questionId,
    assessmentId,
  }
}

export function addFibElement(element, assessmentId) {
  return {
    type: CONSTANTS.ADD_FIB_ELEMENT_REQUEST,
    element,
    assessmentId,
  }
}

export function updateFibElement(elementId, element, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_FIB_ELEMENT_REQUEST,
    elementId,
    element,
    assessmentId,
  }
}

export function deleteFibElement(elementId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_FIB_ELEMENT_REQUEST,
    elementId,
    assessmentId,
  }
}

export const loadSimulatedNodes = nodes => ({
  type: CONSTANTS.LOAD_SIMULATED_NODES_REQUEST,
  nodes,
})

export const setDidUpdate = didUpdate => ({
  type: CONSTANTS.SET_DID_UPDATE_REQUEST,
  didUpdate,
})