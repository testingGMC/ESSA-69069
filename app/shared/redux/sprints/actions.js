/*
 *
 * Sprints actions
 *
 */
import * as CONSTANTS from './constants'

export function fetchSprintsFirstTime(projectId) {
  return {
    type: CONSTANTS.FETCH_SPRINTS_FIRST_TIME_REQUEST,
    projectId,
  }
}
export function fetchSprints(projectId) {
  return {
    type: CONSTANTS.FETCH_SPRINTS_REQUEST,
    projectId,
  }
}

export function addSprint(sprint) {
  return {
    type: CONSTANTS.ADD_SPRINT_REQUEST,
    sprint,
  }
}

export function fetchWorkItems(sprintId) {
  return {
    type: CONSTANTS.FETCH_WORK_ITEMS_REQUEST,
    sprintId,
  }
}

export function addWorkItem(workItem) {
  return {
    type: CONSTANTS.ADD_WORK_ITEM_REQUEST,
    workItem,
  }
}

export function updateWorkItem(sprintId, workItemId, workItem) {
  return {
    type: CONSTANTS.UPDATE_WORK_ITEM_REQUEST,
    oldSprintId: sprintId,
    workItemId,
    workItem,
  }
}

export function deleteWorkItem(sprintId, workItemId) {
  return {
    type: CONSTANTS.DELETE_WORK_ITEM_REQUEST,
    sprintId,
    workItemId,
  }
}

export function updateWorkItemState(sprintId, workItemId, workItem) {
  return {
    type: CONSTANTS.UPDATE_WORK_ITEM_STATE_REQUEST,
    sprintId,
    workItemId,
    workItem,
  }
}

export const setIsAddWorkItemModalVisible = isVisible => ({
  type: CONSTANTS.SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_REQUEST,
  isVisible,
})
export const setIsUpdateWorkItemModalVisible = isVisible => ({
  type: CONSTANTS.SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_REQUEST,
  isVisible,
})
export const setIsAddSprintModalVisible = isVisible => ({
  type: CONSTANTS.SET_IS_ADD_SPRINT_MODAL_VISIBLE_REQUEST,
  isVisible,
})
