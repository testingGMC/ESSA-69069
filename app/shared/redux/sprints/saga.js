import { put, takeEvery, all, call, take } from 'redux-saga/effects'
import * as api from 'shared/services/sprint.service'
import moment from 'moment'
import { FETCH_PROJECT_BY_ID_REQUEST } from 'shared/redux/selectedproject/constants'
import * as CONSTANTS from './constants'
const delay = ms => new Promise(res => setTimeout(res, ms))

// Individual exports for testing
export function* fetchSprintsFirstTime(action) {
  try {
    yield put({
      type: FETCH_PROJECT_BY_ID_REQUEST,
      projectId: action.projectId,
    })
    const data = yield call(api.fetchSprints, action.projectId)

    yield put({
      type: CONSTANTS.FETCH_SPRINTS_FIRST_TIME_SUCCESS,
      sprints: data.model,
    })
    let selectedSprintId
    if (data.model.length) {
      const now = moment().format()
      const pos = data.model.findIndex(
        sprint => sprint.startDate <= now && sprint.dueDate >= now,
      )
      if (pos !== -1) {
        selectedSprintId = data.model[pos].id
      } else {
        selectedSprintId = data.model[data.model.length - 1].id
      }
      yield put({
        type: CONSTANTS.FETCH_WORK_ITEMS_REQUEST,
        sprintId: selectedSprintId,
      })
    }
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_SPRINTS_FIRST_TIME_FAILURE, e })
  }
}

export function* fetchSprints(action) {
  try {
    const data = yield call(api.fetchSprints, action.projectId)
    yield put({
      type: CONSTANTS.FETCH_SPRINTS_SUCCESS,
      sprints: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_SPRINTS_FAILURE, e })
  }
}

export function* addSprint(action) {
  try {
    const data = yield call(api.addSprint, action.sprint)
    yield put({
      type: CONSTANTS.ADD_SPRINT_SUCCESS,
      sprint: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.ADD_SPRINT_FAILURE, e })
  }
}

export function* fetchWorkItems(action) {
  try {
    const data = yield call(api.fetchWorkItems, action.sprintId)

    yield put({
      type: CONSTANTS.FETCH_WORK_ITEMS_SUCCESS,
      sprintId: action.sprintId,
      workItems: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_WORK_ITEMS_FAILURE, e })
  }
}
export function* addWorkItem(action) {
  try {
    const data = yield call(api.addWorkItem, action.workItem)
    yield delay(1000)
    yield put({
      type: CONSTANTS.ADD_WORK_ITEM_SUCCESS,
      sprintId: action.workItem.sprintId,
      workItem: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.ADD_WORK_ITEM_FAILURE, e })
  }
}

export function* updateWorkItem(action) {
  try {
    const data = yield call(
      api.updateWorkItem,
      action.workItemId,
      action.workItem,
    )
    yield put({
      type: CONSTANTS.UPDATE_WORK_ITEM_SUCCESS,
      workItemId: action.workItemId,
      oldSprintId: action.oldSprintId,
      workItem: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.UPDATE_WORK_ITEM_FAILURE, e })
  }
}

export function* updateWorkItemState(action) {
  try {
    const data = yield call(
      api.updateWorkItemState,
      action.workItemId,
      action.workItem,
    )
    yield put({
      type: CONSTANTS.UPDATE_WORK_ITEM_STATE_SUCCESS,
      workItemId: action.workItemId,
      sprintId: action.sprintId,
      workItem: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.UPDATE_WORK_ITEM_STATE_FAILURE, e })
  }
}

export function* deleteWorkItem(action) {
  try {
    yield call(api.deleteWorkItem, action.workItemId)
    yield put({
      type: CONSTANTS.DELETE_WORK_ITEM_SUCCESS,
      sprintId: action.sprintId,
      workItemId: action.workItemId,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.DELETE_WORK_ITEM_FAILURE, e })
  }
}

export function* setIsAddWorkItemModalVisible({ isVisible }) {
  try {
    yield put({
      type: CONSTANTS.SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_SUCCESS,
      isVisible,
    })
  } catch (e) {
    yield put({
      type: CONSTANTS.SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_FAILURE,
    })
  }
}
export function* setIsAddWorkItemModalVisibleWatcher() {
  yield takeEvery(
    CONSTANTS.SET_IS_ADD_WORK_ITEM_MODAL_VISIBLE_REQUEST,
    setIsAddWorkItemModalVisible,
  )
}

export function* setIsUpdateWorkItemModalVisible({ isVisible }) {
  try {
    yield put({
      type: CONSTANTS.SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_SUCCESS,
      isVisible,
    })
  } catch (e) {
    yield put({
      type: CONSTANTS.SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_FAILURE,
    })
  }
}
export function* setIsUpdateWorkItemModalVisibleWatcher() {
  yield takeEvery(
    CONSTANTS.SET_IS_UPDATE_WORK_ITEM_MODAL_VISIBLE_REQUEST,
    setIsUpdateWorkItemModalVisible,
  )
}

export function* setIsAddSprintModalVisible({ isVisible }) {
  try {
    yield put({
      type: CONSTANTS.SET_IS_ADD_SPRINT_MODAL_VISIBLE_SUCCESS,
      isVisible,
    })
  } catch (e) {
    yield put({
      type: CONSTANTS.SET_IS_ADD_SPRINT_MODAL_VISIBLE_FAILURE,
    })
  }
}
export function* setIsAddSprintModalVisibleWatcher() {
  yield takeEvery(
    CONSTANTS.SET_IS_ADD_SPRINT_MODAL_VISIBLE_REQUEST,
    setIsAddSprintModalVisible,
  )
}

export function* fetchSprintsFirstTimeWatcher() {
  yield takeEvery(
    CONSTANTS.FETCH_SPRINTS_FIRST_TIME_REQUEST,
    fetchSprintsFirstTime,
  )
}
export function* fetchSprintsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_SPRINTS_REQUEST, fetchSprints)
}
export function* addSprintWatcher() {
  yield takeEvery(CONSTANTS.ADD_SPRINT_REQUEST, addSprint)
}

export function* fetchWorkItemsWatcher() {
  // yield take(FETCH_PROJECT_MEMBERS_BY_PROJECT_ID_SUCCESS)
  yield takeEvery(CONSTANTS.FETCH_WORK_ITEMS_REQUEST, fetchWorkItems)
}
export function* addWorkItemWatcher() {
  yield takeEvery(CONSTANTS.ADD_WORK_ITEM_REQUEST, addWorkItem)
}

export function* updateWorkItemWatcher() {
  yield takeEvery(CONSTANTS.UPDATE_WORK_ITEM_REQUEST, updateWorkItem)
}

export function* deleteWorkItemWatcher() {
  yield takeEvery(CONSTANTS.DELETE_WORK_ITEM_REQUEST, deleteWorkItem)
}

export function* updateWorkItemStateWatcher() {
  yield takeEvery(CONSTANTS.UPDATE_WORK_ITEM_STATE_REQUEST, updateWorkItemState)
}

export default function* sprintsSaga() {
  yield all([
    fetchSprintsFirstTimeWatcher(),
    addSprintWatcher(),
    fetchSprintsWatcher(),
    fetchWorkItemsWatcher(),
    addWorkItemWatcher(),
    updateWorkItemWatcher(),
    deleteWorkItemWatcher(),
    updateWorkItemStateWatcher(),
    setIsAddWorkItemModalVisibleWatcher(),
    setIsUpdateWorkItemModalVisibleWatcher(),
    setIsAddSprintModalVisibleWatcher(),
  ])
}
