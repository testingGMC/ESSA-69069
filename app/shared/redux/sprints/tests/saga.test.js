/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeEvery } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as actions from '../actions'
import * as CONSTANTS from '../constants'

describe('fetchSprints Saga', () => {
  let fetchSprintsGenerator
  const projectId = '1'
  const action = actions.fetchSprints(projectId)
  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    fetchSprintsGenerator = sagas.fetchSprints(action)
    const callDescriptor = fetchSprintsGenerator.next().value
    expect(callDescriptor).toMatchSnapshot()
  })

  it('should dispatch the FETCH_SPRINTS_SUCCESS action if it requests the data successfully', () => {
    const response = {
      model: [
        { id: '1', name: 'Sprint 1' },
        { id: '2', name: 'Sprint 2' },
        { id: '3', name: 'Sprint 3' },
      ],
    }
    const successAction = {
      type: CONSTANTS.FETCH_SPRINTS_SUCCESS,
      sprints: response.model,
    }
    const putDescriptor = fetchSprintsGenerator.next(response).value
    expect(putDescriptor).toEqual(put(successAction))
  })

  it('should call the FETCH_SPRINTS_FAILURE action if the response errors', () => {
    const response = new Error('Some error')
    const putDescriptor = fetchSprintsGenerator.throw(response).value
    const failureAction = {
      type: CONSTANTS.FETCH_SPRINTS_FAILURE,
    }
    expect(putDescriptor).toEqual(put(failureAction))
  })
})

describe('fetchSprintsWatcher Saga', () => {
  const fetchSprintsWatcher = sagas.fetchSprintsWatcher()

  it('should start task to watch for FETCH_SPRINTS_REQUEST action', () => {
    const takeEveryDescriptor = fetchSprintsWatcher.next().value

    expect(takeEveryDescriptor).toEqual(
      takeEvery(CONSTANTS.FETCH_SPRINTS_REQUEST, sagas.fetchSprints),
    )
  })
})
