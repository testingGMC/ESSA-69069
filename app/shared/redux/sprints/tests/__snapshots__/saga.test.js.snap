// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchSprints Saga should call the FETCH_SPRINTS_FAILURE action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`fetchSprints Saga should dispatch the FETCH_SPRINTS_SUCCESS action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
